shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -72,100 +65,72 @@ public class ManagedBridgeController implements Reconciler<ManagedBridge>,
     @Inject
     OperatorMetricsService metricsService;
 
+    @Inject
+    ManagerClient managerClient;
```

Comment:
I think with v2 we should avoid the usage of `ManagerClient` inside the controllers: what if the manager is not reachable? Do we loose the update? K8s update succeed so it is going to have locally the resource updated but the status on the Manager remains wrong.
I would move also the shard -> manager logic in a scheduled method that periodically send back ~everything (aka all the statuses) to the manager. We can do it smarter with some delta but I would start simple with a full update.
In this way we also preserve the single write principle that makes the system reliable.

@robpblake 
Wdyt?

Reply:
> @danielezonca @vaibhavjainwiz
> 
> > We can do it smarter with some delta but I would start simple with a full update
> 
> The `manager` needs and depends on the full set at the moment.

@manstis Just to clarify, I think the question is can we be smarter about which `ManagedBridge` we send the `conditions` for, not which `conditions` do we send for a particular `ManagedBridge`. It's 100% the intention at this point that the data plane reports all `conditions` on the `ManagedBridge`.

Reply:
> 

@manstis I'm not sure about this. I think the Controller should be the one who would have the knowledge as to why/where the timeout had occurred and should report it via the `Conditions` it adds to the resource. This keeps all provisioning logic for a particular resource encapsulated within the Controller and reported via the Conditions e.g we could decide that we have different timeouts for certain resources.

It also keeps the role of the synchronizer simple: send and receive updates from the Control Plane

We also have https://issues.redhat.com/browse/MGDOBR-1308 to discuss at some point in which we move the general "timeout" back to the control plane.

Reply:
done..
As suggested I had introduced a new `Scheduler` inside `ManagerBridgeSyncService` to send status back.